src/openads/application/service/DisplayPositionUseCase.js

```diff
@@ -29,4 +35,18 @@ export default class DisplayPositionUseCase {
     }
     return optionalPositionWithId.position
   }
+
+  _filterPositionAdAvailable (position) {
+    return position.ad
+      .then(adResponse => adResponse.status)
+      .then(status => AD_AVAILABLE === status)
+      .then(available => available ? position : Promise.reject(new PositionAdNotAvailableError({id: position.id})))
+  }
+
+  _filterPositionAdIsNative (position) {
```

Comment:
@alextremp is right, change the method name by something like _filterPositionAdNoNative

Reply:
It is done!
![](https://media.giphy.com/media/3o7ZeGFFAZUc79Dsxq/giphy.gif)